src/Core/Commander/ManagerCommands.js

```diff
@@ -77,44 +80,35 @@ define('Core/Commander/ManagerCommands', [
             return this.commandsLength()===0;
         };
 
-        ManagerCommands.prototype.runAllCommands = function() {
-
-
-            if (this.commandsLength() === 0)
-            {
-                return Promise.resolve(0);
+        ManagerCommands.prototype.fillCommandPool = function() {
+
+            var command;
+            var that = this;
+            function launchCommand(cmd) {
+                var providers = that.getProviders(cmd.layer);
+                for (var i = 0; i < providers.length; i++) {
+                    that.currentCommandsCount++;
+                    var p = providers[i].executeCommand(cmd);
+                    if(p) {
+                        p.then(function() {
+                            that.currentCommandsCount--;
+                            that.executedCommands++;
```

Comment:
`executedCommands` could have the same `Count` suffix.
Or actually I'd prefer them to leave in a sub-object:
```
this.counters.currentCommand--
this.counters.executedCommands++
```